src/cargo/util/config.rs

```diff
@@ -386,7 +386,7 @@ impl Config {
         let cfg_verbose = self.get_bool("term.verbose").unwrap_or(None).map(|v| v.val);
         let cfg_color = self.get_string("term.color").unwrap_or(None).map(|v| v.val);
 
-        let color = color.as_ref().or(cfg_color.as_ref());
+        let color = color.as_ref().or_else(|| cfg_color.as_ref());
```

Comment:
there are a bunch of `.foo(|| someFunc())` in this PR - can any of them be rewritten by passing the function directly: `.foo(someFunc)`? In this case, that would be `color.as_ref().or_else(cfg_color.as_ref)`.

Note that in this specific case something even better is possible: `color.or(cfg_color).as_ref()`

Reply:
You are right, that was an oversight. I have many more changes waiting and will incorporate this in the next PR.